text/0000-swap-out-jemalloc.md

```diff
@@ -0,0 +1,235 @@
+- Feature Name: `allocator`
+- Start Date: 2015-06-27
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Add support to the compiler to override the default allocator, allowing a
+different allocator to be used by default in Rust programs. Additionally, also
+switch the default allocator for dynamic libraries and static libraries to using
+the system malloc instead of jemalloc.
+
+# Motivation
+
+Note that this issue was [discussed quite a bit][babysteps] in the past, and
+the meat of this RFC draws from Niko's post.
+
+[babysteps]: http://smallcultfollowing.com/babysteps/blog/2014/11/14/allocators-in-rust/
+
+Currently all Rust programs by default use jemalloc for an allocator because it
+is a fairly reasonable default as it is commonly much faster than the default
+system allocator. This is not desirable, however, when embedding Rust code into
+other runtimes. Using jemalloc implies that Rust will be using one allocator
+while the host application (e.g. Ruby, Firefox, etc) will be using a separate
+allocator. Having two allocators in one process generally hurts performance and
+is not recommended, so the Rust toolchain needs to provide a method to configure
+the allocator.
+
+In addition to using an entirely separate allocator altogether, some Rust
+programs may want to simply instrument allocations or shim in additional
+functionality (such as memory tracking statistics). This is currently quite
+difficult to do, and would be accomodated with a custom allocation scheme.
+
+# Detailed design
+
+The high level design can be found [in this gist][gist], but this RFC intends to
+expound on the idea to make it more concrete in terms of what the compiler
+implementation will look like. A [sample implementaiton][impl] is available of
+this section.
+
+[gist]: https://gist.github.com/alexcrichton/41c6aad500e56f49abda
+[impl]: https://github.com/alexcrichton/rust/tree/less-jemalloc
+
+### High level design
+
+The design of this RFC from 10,000 feet (referred to below), which was
```

Comment:
Aviation still uses feet for altitude in most of the world, so let the idiom be! :stuck_out_tongue: